src/meta/app/src/background/background_job.rs

```diff
@@ -0,0 +1,417 @@
+// Copyright 2021 Datafuse Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+use std::fmt::Display;
+use std::fmt::Formatter;
+use std::str::FromStr;
+
+use chrono::DateTime;
+use chrono::Utc;
+use cron::Schedule;
+
+use crate::background::BackgroundTaskType;
+use crate::principal::UserIdentity;
+
+#[derive(
+    serde::Serialize,
+    serde::Deserialize,
+    Clone,
+    Debug,
+    Default,
+    Eq,
+    PartialEq,
+    num_derive::FromPrimitive,
+)]
+pub enum BackgroundJobState {
+    #[default]
+    RUNNING = 0,
+    FAILED = 1,
+    SUSPENDED = 2,
+}
+
+impl Display for BackgroundJobState {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match self {
+            BackgroundJobState::RUNNING => write!(f, "RUNNING"),
+            BackgroundJobState::FAILED => write!(f, "FAILED"),
+            BackgroundJobState::SUSPENDED => write!(f, "SUSPENDED"),
+        }
+    }
+}
```

Comment:
It's not a problem at all but such implementation can be written as
```suggestion
impl Display for BackgroundJobState {
    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
        write!(f, "{:?}", self)
    }
}
```

Because the `Debug` formatter just outputs the name of the variant.

Or you can just use crate `derive_more`:
```rust
#[derive(derive_more::Display)]
enum BackgroundJobState {
...
}
```